### PR TITLE
Jason's changes - WIP

### PR DESCRIPTION
**Pull request by jasonmckenna**
Changes required to compile on my machine, and start divinding into Game/PlayerAbstract/PlayerLEDDriver layers

Classes added
Game - holds game state and global table state
Player - holds per-player data (colour, where in their personal LED sequence they are, which segment of LEDs are theirs
Led - interfaces with FastLED to drive a subset of LED strip

### CODE REVIEW

File: include/game.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#ifndef __GAME_H
```
Comment (jasonmckenna) [inline include/game.h:1]: The idea of this is that the Game class will hold state data about the game in progress (how many players there are, the per-player data, the global game mode, etc).  Cleaner than having a butt-load of global variables

File: include/ledRunner.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+typedef enum {
+    LED_MODE__OFF = 0,
+    LED_MODE__SOLID,
+    LED_MODE__BLINK,
+    LED_MODE__BREATH,
+    LED_MODE__COUNTDOWN,
+    LED_MODE__NUMMODES
+} ledMode_t;
+
+#endif /* __LED_RUNNER_H */
```
Comment (jasonmckenna) [inline include/ledRunner.h:15]: Placeholder -- will eventually have a class that will drive a giver player's LEDs according to any number of differnt modes

File: include/player.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        int ledState;
+        ledMode_t ledMode;
+        CRGB colour;
+    public:
+        Player(int startLed, int endLed, CRGB colour);
+        virtual ~Player();
+        virtual void setLedMode(ledMode_t mode);
+};
+
+#endif /* __PLAYER_H */
```
Comment (jasonmckenna) [inline include/player.h:19]: Hold per-player info - where their LEDs are in the global array, what colour the player is, what mode their LEDs are running in, etc

File: platformio.ini [hunk 4]
```diff
@@ -15,13 +15,8 @@ framework = arduino
 monitor_speed = 115200
 test_port = COM3
 lib_deps = 
-    fastled/FastLED @ ^3.4.0
-    https://github.com/bigsdawg-2k/command_buffer.git
-
-; Debug tool = ESP-PROG
+	https://github.com/bigsdawg-2k/command_buffer.git
+	fastled/FastLED @ ^3.5.0
```
Comment (jasonmckenna) [inline platformio.ini:19]: For whatever reason FastLED 3.4.0 doesn't compile with the esp32 Arduino headers that platformIO pulled down.  I needed to update to 3.5.0

File: src/main.cpp [hunk 5]
```diff
@@ -487,3 +502,6 @@ void loop()
  
 }
 
+int processCtrlCmd(void) {
+    return 0;
+}
```
Comment (jasonmckenna) [inline src/main.cpp:507]: Moved implementation from header to .c

**Review by bigsdawg-2k — APPROVED** [review]
Thanks for setting this up.  I did start making changes and commit things that were definitely barely implemented.

**Pull request merged**